src/kbc_modules/cc_kbc/mod.rs

```diff
@@ -94,4 +127,86 @@ impl Kbc {
             .ok_or_else(|| anyhow!("TEE rsa key missing"))?;
         response.decrypt_output(key)
     }
+
+    fn tee(&self) -> &str {
+        &self.tee
+    }
+
+    fn kbs_uri(&self) -> &str {
+        &self.kbs_uri
+    }
+
+    fn http_client(&mut self) -> &mut reqwest::Client {
+        &mut self.http_client
+    }
+
+    async fn establish_kbs_session(&mut self) -> Result<()> {
+        let kbs_uri = self.kbs_uri().to_string();
+
+        let challenge = self
+            .http_client()
+            .post(format!("{}/auth", kbs_uri))
+            .header("Content-Type", "application/json")
+            .json(&Request::new(self.tee().to_string()))
+            .send()
+            .await?
+            .json::<Challenge>()
+            .await?;
+        self.nonce = challenge.nonce.clone();
+
+        let attest_response = self
+            .http_client()
+            .post(format!("{}/attest", kbs_uri))
+            .header("Content-Type", "application/json")
+            .json(&self.generate_evidence()?)
+            .send()
+            .await?;
+        if attest_response.status() != reqwest::StatusCode::OK {
+            // TODO: Parse Error description payload
+            bail!("KBS attest failed")
+        }
+
+        self.authenticated = true;
+
+        Ok(())
+    }
+
+    async fn request_kbs_resource(&mut self, resource_url: String) -> Result<Response> {
+        for attempt in 0..KBS_GET_RESOURCE_MAX_ATTEMPT {
+            log::info!("CC-KBC: trying to get resource, attempt {attempt}");
+
+            if !self.authenticated {
+                self.establish_kbs_session().await?;
+            }
+
+            let res = self.http_client().get(&resource_url).send().await?;
+
+            if res.status() == reqwest::StatusCode::UNAUTHORIZED {
+                self.authenticated = false;
+                continue;
+            }
+
+            if res.status() == reqwest::StatusCode::OK {
+                let response = res.json::<Response>().await?;
+                return Ok(response);
+            } else {
+                // TODO: Parse Error description payload
+                bail!("Request KBS resource failed")
+            }
```

Comment:
[style] consider changing these `if`s and `else` into a `match req.status() { ... }`.

Reply:
Done!